### PR TITLE
Add LLM fallback capability.

### PR DESCRIPTION
**Pull request by stevewallone**
This PR implements automatic LLM fallback functionality to improve system reliability when the primary LLM becomes unavailable. The implementation follows an MVP approach focused on simplicity, allowing the system to immediately switch to a fallback LLM on any primary failure, with automatic recovery detection.

In response to the feedback about implementing fallback at the workflow level, a different
approach was taken where the LLM interaction is encapsulated in a singleton LLMManager, which was accessed through the activity requiring that functionality. The thought was around cohesion and coupling. Where the LLMManager is a cohesive component containing the logic needed for LLM interaction. Keeping this interaction within the activity reduces coupling with the workflow such that the workflow remains ignorant of the inner workings of a given activity.

- Single Point of Control: All LLM logic is centralized in the LLMManager class
- Immediate Fallback: No retries or delays - switch immediately on any failure
- Transparent Integration: Existing code continues to work with minimal changes
- MVP First: Start simple, evolve based on real-world experience

This approach ensures maximum availability while keeping the complexity manageable for an MVP implementation.

Key Changes

🆕 New Core Component: LLMManager (shared/llm_manager.py)

- New functionality implementing singleton-pattern LLM management
- Immediate fallback switching on any primary LLM failure
- Automatic recovery detection with configurable check intervals
- Comprehensive logging and status reporting for debugging
- Environment-driven configuration with sensible defaults
- Configurable timeouts for both primary and fallback LLMs

Key Features:
- Singleton pattern ensures consistent state across the application
- Health check mechanism to detect when primary LLM recovers
- Comprehensive status reporting for monitoring and debugging
- Optional debug file output for troubleshooting

🔧 Enhanced Tool Activities (activities/tool_activities.py)

- Integrated LLMManager for automatic fallback support
- Function name correction: agent_toolPlanner → agent_tool_planner (Python naming convention)
- Added Ollama warm-up functionality to reduce cold start latency
- Enhanced error logging with LLM status information

🌍 Environment Configuration (.env.example)

New Environment Variables:
# Primary LLM Configuration
LLM_TIMEOUT_SECONDS=10                    # Configurable timeout
LLM_DEBUG_OUTPUT=false                    # Debug file output

# Fallback LLM Configuration
LLM_FALLBACK_MODEL=openai/gpt-4o          # Fallback model
LLM_FALLBACK_KEY=sk-proj-...              # Fallback API key
LLM_FALLBACK_BASE_URL=...                 # Optional custom endpoint
LLM_FALLBACK_TIMEOUT_SECONDS=10           # Fallback timeout
LLM_RECOVERY_CHECK_INTERVAL_MINUTES=5     # Recovery check frequency

🛠️ Development Tools (dev-tools/)

New Packet Filter Management Scripts:
- allow-anthropic.sh  - Allow Anthropic API access
- block-anthropic.sh  - Block Anthropic API access

These tools help with testing fallback scenarios by controlling network access to specific LLM providers.

🧪 Comprehensive Testing (tests/)

New Test Suite: test_llm_manager.py
- 38 comprehensive unit tests covering all LLMManager functionality
- 100% method coverage (excluding debug file methods as requested)
- All function calls properly mocked for isolation
- Edge case testing including timeout configurations, singleton behavior, and error handling

Updated Existing Tests:
- Modified activity and tool activities tests to accommodate new LLM integration

📝 Documentation Updates

- Added TOC to .md files
- docs/setup.md: Enhanced setup instructions

### CODE REVIEW

File: activities/tool_activities.py [hunk 1]
```diff
@@ -101,15 +105,15 @@ async def agent_validatePrompt(
...
+        result = await self.agent_tool_planner(prompt_input)
 
         return ValidationResult(
             validationResult=result.get("validationResult", False),
             validationFailedReason=result.get("validationFailedReason", {}),
         )
 
     @activity.defn
-    async def agent_toolPlanner(self, input: ToolPromptInput) -> dict:
+    async def agent_tool_planner(self, input: ToolPromptInput) -> dict:
```
Comment (steveandroulakis) [inline activities/tool_activities.py:116]: This has been changed but agent_validatePrompt has not, introducing an inconsistency

**Comment by steveandroulakis** [comment]
Thank you. I need to think about this! This may be a big question of "what is this sample _for_?".

The singleton method means the workflow never “sees” an outage as that logic is pushed down into the activity itself, so the history/UI just shows a normal activity completion. **That hides exactly the durable, observable failure demo story I've been trying to showcase with Temporal.** I realize I can firewall/block things out, incl with your helper scripts, but there was charm to the fact these APIs often are flaky in and of themselves and Temporal visibly (UI) is seen recovering like it's nothing. And I realize there are worker logs, but that's different from a demo perspective.

The singleton approach is neat, but is also localized to a single worker. Maybe that's not bad, as each would have to fallback (and mightn't) but I don't know. There's a big observability story in Temporal + agents that I don't wanna miss.

I'll evaluate this properly in the next 2wks and get back.

File: activities/tool_activities.py (hunk 1 above)
Comment (stevewallone) [reply activities/tool_activities.py:116]: Good point. I could resolve this in this PR or possibly create another one immediately after to address the inconsistency.

**Comment by stevewallone** [comment]
I think I see what you are getting at in that does this PR address the driving requirement of having a demo for durability. Maybe a deeper conversation would be useful to get better alignment. I'll head over to Slack and pick it up there.

**Comment by stevewallone** [comment]
These enhancement were not inline the fundamental vision. Another branch was created and the appropriate changes will be made there.
